crates/edr_solidity/src/contracts_identifier.rs

```diff
@@ -0,0 +1,107 @@
+mod radix_tree;
+
+use std::collections::hash_map::DefaultHasher;
+use std::collections::HashMap;
+use std::hash::{Hash, Hasher};
+
+use radix_tree::RadixTree;
+
+// bytecode and trace stubs
+pub struct Bytecode {
+    normalized_code: Vec<u8>,
+}
+
+pub struct CreateMessageTrace {
+    code: Vec<u8>,
+}
+
+pub struct CallMessageTrace {
+    code: Vec<u8>,
+}
+
+pub enum EvmMessageTrace {
+    Create(CreateMessageTrace),
+    Call(CallMessageTrace),
+}
+
+impl EvmMessageTrace {
+    fn get_code(&self) -> &Vec<u8> {
+        match self {
+            EvmMessageTrace::Create(create_message_trace) => &create_message_trace.code,
+            EvmMessageTrace::Call(call_message_trace) => &call_message_trace.code,
+        }
+    }
+}
+
+// TODO add cache
+pub struct ContractsIdentifier<'a> {
+    tree: RadixTree,
+    bytecodes: HashMap<u64, &'a Bytecode>,
+}
+
+impl<'a> ContractsIdentifier<'a> {
+    pub fn new() -> Self {
+        Self {
+            tree: RadixTree::new(),
+            bytecodes: HashMap::new(),
+        }
+    }
```

Comment:
Clippy should have complained about this (are you running it?), but the `Default` implement should be used instead of an empty constructor

Reply:
Done.